components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabsToolbarFeature.kt

```diff
@@ -92,7 +92,16 @@ class CustomTabsToolbarFeature(
         }.also {
             val button = Toolbar.ActionButton(
                 it, context.getString(R.string.mozac_feature_customtabs_exit_button)
-            ) { closeListener.invoke() }
+            ) {
+                val selectedId = sessionManager.selectedSession?.id
+                for (i in 0..sessionManager.sessions.size) {
+                    if (sessionManager.sessions[i].id == selectedId) {
+                        emitCloseFact(i, sessionManager.sessions.size)
+                    }
+                }
```

Comment:
FYI: There's also a `findSessionById()` method.

I wonder what this does though. It emits a close fact here, but it doesn't actually close this session here. Is this the right place?

Reply:
This should get called right before `closeListener.invoke()` is called right beneath it, no? We care that the "close" button is pressed, not that the session is closed.

Also, I need to know the **index** of the session, which is why I chose to use a for loop rather than `findSessionById()`.

Reply:
I'm not really sure what you are trying to achieve here though: This is a custom tab that we are closing: It is not `selectedSession` and its also not part of `sessions`. Why do those values matter here? I can understand that they are interesting in the context of an actual tab getting closed. :)

Reply:
I raised these concerns [in the original ticket](https://github.com/mozilla-mobile/fenix/issues/977)!

Reply:
@pocmo I updated the PR to remove this, as product decided that detail wasn't necessary. Thanks for bringing this up 😄